ui/src/components/App.js

```diff
@@ -21,65 +18,105 @@ import HelpContainer from '../containers/HelpContainer';
 import Main from './Main';
 import LoadingScreen from '../components/LoadingScreen/LoadingScreen';
 
-async function requireAuth() {
-  await store.dispatch(getLoginInfo());
-  const { login } = store.getState();
+import { store } from '../store';
+import { serverIO } from '../serverIO';
 
-  if (login.loggedIn) {
-    await store.dispatch(startSession(login.user.inControl));
-  }
+import { getLoginInfo } from '../actions/login';
+import { getInitialState, applicationFetched } from '../actions/general';
 
-  if (login.loggedIn && !serverIO.initialized) {
-    serverIO.listen(store);
+export async function requireAuth() {
+  try {
+    store.dispatch(applicationFetched(false));
+    await store.dispatch(getLoginInfo());
+  } catch {
+    store.dispatch(applicationFetched(true));
+    return;
```

Comment:
If `getLoginInfo` fails, then the app shouldn't remain in a loading state. I think this is how the app behaves currently but it's not great, obviously. Might be good to fix this behaviour as part of this PR.

Reply:
I think its solved with the ` store.dispatch(applicationFetched(true));`  application is considered fetched if `getLoginInfo` fails in the code above or when `getInitialState()` is done.

Reply:
Oh right, I must have misread sorry. The error will need to be handled a bit better in a future PR, then. :+1: